src/Components/Components/test/RendererTest.cs

```diff
@@ -4970,5 +5202,90 @@ public Task SetParametersAsync(ParameterView parameters)
                 return new TaskCompletionSource<object>().Task;
             }
         }
+
+        private class TestErrorBoundary : AutoRenderComponent, IErrorBoundary
+        {
+            private TaskCompletionSource receivedErrorTaskCompletionSource = new();
+
+            public Exception ReceivedException { get; private set; }
+            public Task ReceivedErrorTask => receivedErrorTaskCompletionSource.Task;
+
+            [Parameter] public RenderFragment ChildContent { get; set; }
+
+            protected override void BuildRenderTree(RenderTreeBuilder builder)
+                => ChildContent(builder);
+
+            public void HandleException(Exception error)
+            {
+                ReceivedException = error;
+                receivedErrorTaskCompletionSource.SetResult();
+            }
+
+            public static void RenderNestedErrorBoundaries(RenderTreeBuilder builder, RenderFragment innerContent)
+            {
+                // Create an error boundary
+                builder.OpenComponent<TestErrorBoundary>(0);
+                builder.AddAttribute(1, nameof(TestErrorBoundary.ChildContent), (RenderFragment)(builder =>
+                {
+                    // ... containing another error boundary, containing the content
+                    builder.OpenComponent<TestErrorBoundary>(0);
+                    builder.AddAttribute(1, nameof(TestErrorBoundary.ChildContent), innerContent);
+                    builder.CloseComponent();
+                }));
+                builder.CloseComponent();
+            }
+        }
+
+        private class ErrorThrowingComponent : AutoRenderComponent, IHandleEvent
+        {
+            [Parameter] public Exception ThrowDuringRender { get; set; }
+            [Parameter] public Exception ThrowDuringEventSync { get; set; }
+            [Parameter] public Exception ThrowDuringEventAsync { get; set; }
+            [Parameter] public Exception ThrowDuringParameterSettingSync { get; set; }
+            [Parameter] public Exception ThrowDuringParameterSettingAsync { get; set; }
+
+            public override async Task SetParametersAsync(ParameterView parameters)
+            {
+                _ = base.SetParametersAsync(parameters);
+
+                if (ThrowDuringParameterSettingSync is not null)
+                {
+                    throw ThrowDuringParameterSettingSync;
+                }
+
+                if (ThrowDuringParameterSettingAsync is not null)
+                {
+                    await Task.Delay(100);
```

Comment:
This is likely going to cause flackyness.

Is there a way we can do this with a TaskCompletionSource that we pass down from the test?
That's how we do it in other tests on this class for that reason.

Reply:
Good point, fixed